### PR TITLE
Thomas Explore Events Calendar

### PR DESCRIPTION
**Pull request by Tcintra**
Implement calendar structure to explore events with search, filter, and navigation functionalities. Some style fixes in events page.

### CODE REVIEW

**Review by rivera-fernando — APPROVED** [review]

**Comment by Tcintra** [comment]
This PR also contains a bunch of fixes for authentication using HttpSessions!

**Review by christinachance — APPROVED** [review]

File: project/src/main/java/com/google/sps/classes/Day.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+package com.google.sps.classes;
+
+import com.google.sps.classes.Event;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+ 
+/**
+ * Class representing an individual day with its constituent events.
+ *
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/classes/Day.java]: Remove line

File: project/src/main/java/com/google/sps/classes/Event.java [hunk 2]
```diff
@@ -32,8 +32,11 @@
   private final boolean isMine;
   private final String entityType;
   private final String imageKey;
+  private final int day;
+  private final int numAttendees;
+  private final int maxCapacity;
  
-  public Event(long id, String name, String location, String date, String time, String description, String attendance, String type, long timeStamp, boolean isMine, String entityType, String imageKey) {
+  public Event(long id, String name, String location, String date, String time, String description, String attendance, String type, long timeStamp, boolean isMine, String entityType, String imageKey, int day, int numAttendees, int maxCapacity) {
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/classes/Event.java]: Could you make this line not too long. Usually we have threshold of 80 chars per line

File: project/src/main/java/com/google/sps/servlets/Dashboard.java [hunk 3]
```diff
@@ -48,19 +48,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-        String person=(String)session.getAttribute("person"); 
-        String isAdmin = person.substring(person.indexOf("admin\":")+7, person.indexOf("}"));
-        if (person.equals("null")) {
-            response.sendRedirect("/login.html");
-        }
-        else if (isAdmin.equals("true")) {
-            response.sendRedirect("/admin_dash.html");
-        }
-        else {
+    Boolean found = false;
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/Dashboard.java]: boolean?

File: project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java [hunk 4]
```diff
@@ -75,7 +78,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             long timestamp = (long) entity.getProperty("timestamp");
             String imageKey = (String) entity.getProperty("imageKey");
 
-            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, true, "PastEvent", imageKey);
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, true, "PastEvent", imageKey, -1, attendees.size(), 0);
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java]: Keep 80 char line size

File: project/src/main/java/com/google/sps/servlets/LoadUpcomingEventsServlet.java [hunk 5]
```diff
@@ -75,7 +78,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             long timestamp = (long) entity.getProperty("timestamp");
             String imageKey = (String) entity.getProperty("imageKey");
  
-            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent", imageKey);
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent", imageKey, -1, attendees.size(), 0);
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LoadUpcomingEventsServlet.java]: 80 characters line size

Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/LoadUpcomingEventsServlet.java]: done!

File: project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java (hunk 4 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/LoadPastEventsServlet.java]: done!

File: project/src/main/java/com/google/sps/servlets/Dashboard.java (hunk 3 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/Dashboard.java]: done!

File: project/src/main/java/com/google/sps/classes/Event.java (hunk 2 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/classes/Event.java]: done!

File: project/src/main/java/com/google/sps/classes/Day.java (hunk 1 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/classes/Day.java]: done!

File: project/src/main/java/com/google/sps/servlets/LoadExploreServlet.java [hunk 6]
```diff
@@ -74,19 +93,92 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+            if (dateTimestamp < firstDayOfWeek - 60000) {
+              continue;
+            } else if (dateTimestamp >= firstDayOfWeek + 604740000) {
+              break;
+            }
+            Calendar calendar = Calendar.getInstance();
+            calendar.clear();
+            calendar.setTimeInMillis(dateTimestamp);
+            int day = calendar.get(Calendar.DAY_OF_WEEK);
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent", imageKey, day, attendees.size(), 0);
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LoadExploreServlet.java:106]: 80 char line please

File: project/src/main/java/com/google/sps/servlets/LoadPendingEventsServlet.java [hunk 7]
```diff
@@ -94,7 +97,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             boolean isMine =  entity.getProperty("email").equals(email);
             String imageKey = (String) entity.getProperty("imageKey");
  
-            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, true, "UnapprovedEvent", imageKey);
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, true, "UnapprovedEvent", imageKey, -1, 0, 0);
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LoadPendingEventsServlet.java:100]: 80 char line please

File: project/src/main/java/com/google/sps/servlets/SearchEventsServlet.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+            String attendance = (String) entity.getProperty("type");
+            long timestamp = (long) entity.getProperty("timestamp");
+            String imageKey = (String) entity.getProperty("imageKey");
+            long dateTimestamp = (long) entity.getProperty("dateTimestamp");
+
+            Calendar calendar = Calendar.getInstance();
+            calendar.clear();
+            calendar.setTimeInMillis(dateTimestamp);
+            int day = calendar.get(Calendar.DAY_OF_WEEK);
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent", imageKey, day, attendees.size(), 0);
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/SearchEventsServlet.java:91]: 80 char line please

**Review by serzhk — APPROVED** [review]
Fix 80 char lines please and merge it

**Pull request merged**